sklearn/gaussian_process/_tpr.py

```diff
@@ -0,0 +1,441 @@
+        Target values in training data (also required for prediction).
+
+    kernel_ : kernel instance
+        The kernel used for prediction. The structure of the kernel is the
+        same as the one passed as parameter but with optimized hyperparameters.
+
+    L_ : array-like of shape (n_samples, n_samples)
+        Lower-triangular Cholesky decomposition of the kernel in ``X_train_``.
+
+    alpha_ : array-like of shape (n_samples,)
+        Dual coefficients of training data points in kernel space.
+
+    log_marginal_likelihood_value_ : float
+        The log-marginal-likelihood of ``self.kernel_.theta``.
+
+    n_features_in_ : int
+        Number of features seen during :term:`fit`.
+
+    feature_names_in_ : ndarray of shape (`n_features_in_`,)
+        Names of features seen during :term:`fit`. Defined only when `X`
+        has feature names that are all strings.
+
+    See Also
+    --------
+    GaussianProcessClassifier : Gaussian process regression (GPR)
+
+    References
+    ----------
+    .. [SW2014] `Amar Shah, Andrew Gordon Wilson, Zoubin Ghahramani,
+       "Student-t Processes as Alternatives to Gaussian Processes",
+       arxiv > stat > arXiv:1402.4306 <https://arxiv.org/abs/1402.4306>`_
+
+    .. [TW2018] `Brendan D. Tracey, David H. Wolpert,
+       "Upgrading from Gaussian Processes to Student's-T Processes"
+        arxiv > stat > arXiv:1801.06147 <https://arxiv.org/abs/1801.06147>`_
+
+
+    .. [RW2006] `Carl E. Rasmussen and Christopher K.I. Williams,
+       "Gaussian Processes for Machine Learning",
+       MIT Press 2006 <https://www.gaussianprocess.org/gpml/chapters/RW.pdf>`_
+
+    Examples
+    --------
+    >>> from sklearn.datasets import make_friedman2
+    >>> from sklearn.gaussian_process import TProcessRegressor
+    >>> from sklearn.gaussian_process.kernels import DotProduct, WhiteKernel
+    >>> X, y = make_friedman2(n_samples=500, noise=0, random_state=0)
+    >>> kernel = DotProduct() + WhiteKernel()
+    >>> tpr = TProcessRegressor(kernel=kernel,
+    ...         random_state=0).fit(X, y)
+    >>> tpr.score(X, y)
+    0.8690...
+    >>> tpr.predict(X[:2,:], return_std=True)
+    (array([[754.5..., 526.2...]), array([147.8..., 148.0...]))
+    """
+
+    def __init__(
+        self,
+        kernel=None,
+        v=3,
+        *,
+        alpha=1e-10,
+        optimizer="fmin_l_bfgs_b",
+        n_restarts_optimizer=0,
+        normalize_y=False,
+        copy_X_train=True,
+        n_targets=None,
+        random_state=None,
+    ):
+        super().__init__(
+            kernel,
+            alpha=alpha,
+            optimizer=optimizer,
+            n_restarts_optimizer=n_restarts_optimizer,
+            normalize_y=normalize_y,
+            copy_X_train=copy_X_train,
+            n_targets=n_targets,
+            random_state=random_state,
+        )
+        self.m_dis = 0  # Mahalanobis Distance
+        self.v0 = v  # Starting degrees of freedom
```

Comment:
May be good to add a comment specifying that degrees of freedom should be more than 2

Reply:
Ya, it not in there already would be worth raising an error if df < 2